src/mds/CInode.cc

```diff
@@ -4110,35 +4111,14 @@ int CInode::encode_inodestat(bufferlist& bl, Session *session,
    */
   if (session->info.has_feature(CEPHFS_FEATURE_REPLY_ENCODING)) {
     ENCODE_START(7, 1, bl);
-    encode(oi->ino, bl);
-    encode(snapid, bl);
-    encode(oi->rdev, bl);
-    encode(version, bl);
-    encode(xattr_version, bl);
+    encode(std::tuple{oi->ino, snapid, oi->rdev, version, xattr_version}, bl, 0);
     encode(ecap, bl);
     {
       ceph_file_layout legacy_layout;
       layout.to_legacy(&legacy_layout);
       encode(legacy_layout, bl);
     }
-    encode(any_i->ctime, bl);
-    encode(file_i->mtime, bl);
-    encode(file_i->atime, bl);
-    encode(file_i->time_warp_seq, bl);
-    encode(file_i->size, bl);
-    encode(max_size, bl);
-    encode(file_i->truncate_size, bl);
-    encode(file_i->truncate_seq, bl);
-    encode(auth_i->mode, bl);
-    encode((uint32_t)auth_i->uid, bl);
-    encode((uint32_t)auth_i->gid, bl);
-    encode(link_i->nlink, bl);
-    encode(file_i->dirstat.nfiles, bl);
-    encode(file_i->dirstat.nsubdirs, bl);
-    encode(file_i->rstat.rbytes, bl);
-    encode(file_i->rstat.rfiles, bl);
-    encode(file_i->rstat.rsubdirs, bl);
-    encode(file_i->rstat.rctime, bl);
+    encode(std::tuple{any_i->ctime, file_i->mtime, file_i->atime, file_i->time_warp_seq, file_i->size, max_size, file_i->truncate_size, file_i->truncate_seq, auth_i->mode, (uint32_t)auth_i->uid, (uint32_t)auth_i->gid, link_i->nlink, file_i->dirstat.nfiles, file_i->dirstat.nsubdirs, file_i->rstat.rbytes, file_i->rstat.rfiles, file_i->rstat.rsubdirs, file_i->rstat.rctime}, bl, 0);
```

Comment:
Interesting! I like it but would prefer this be formatted with one item per line. Perhaps also:

```
auto t = std::tuple{
foo,
bar,
};
encode(std::move(t), bl, 0);
```

?

Reply:
If you give this variable a name, then this might create more code and create runtime overhead because it will be deallocated/destructed at the end of the block; using an anonymous temporary is always better because it is guaranteed to be deallocated+destructed at the end of this one statement.
It might be optimized away, but that is not guaranteed.
Additionally, if it gets passed by reference, the compiler might not be able to prove that the reference is never leaked and might be forced to postpone stack deallocation, not being able to reuse that part of the stack immediately.

Reply:
Not really sure how significant a concern that is. You can get finer control with a new block over the allocation.

I'd still like the formatting to be improved by splitting each member across lines:

```
encode(std::tuple{
  any_i->ctime,
  ...
  }, bl, 0);
```

but that is pretty ugly. It will still read better.

Reply:
Reformatted. With trailing comma so new lines can be added easily.